src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace OSPSuite.InstallationValidator.Core.Domain
+{
+   public class StartableProcess
+   {
+      private readonly Process _process;
+      private bool _exited;
+
+      public StartableProcess(string filePath, string arguments = null)
+      {
+         _process = new Process
+         {
+            StartInfo = new ProcessStartInfo
+            {
+               FileName = filePath,
+               Arguments = arguments ?? string.Empty
+            }
+         };
+         subscribe();
+      }
+
+      /// <summary>
+      /// <returns>true</returns> if the process exited normally. If the process was stopped, returns false
+      /// </summary>
+      public bool ExitedNormally => _exited;
+
+      /// <summary>
+      /// <returns>the exit code of the process</returns> if <see cref="ExitedNormally"/> returns true.
+      /// throws an <seealso cref="InvalidOperationException"/> if the process has not exited or was aborted abnormally
+      /// </summary>
+      public int ReturnCode => _process.ExitCode;
+
+      public virtual void Start()
+      {
+         _process?.Start();
+      }
+
+      public virtual void Stop()
+      {
+         if (_process == null || _exited)
+            return;
+
+         unsubscribe();
+         _process.Kill();
+         _process.Dispose();
```

Comment:
It is never disposed if you do not call Stop explicitly. At least I did not see how